[WIP] Hide the 'share this' section in the print post view.

spsiddarthan opened the pull request:
Fixes https://github.com/Automattic/jetpack/issues/16205

Jetpack should automatically hide extraneous elements like Likes, Sharing, and Related posts in print media. These are interactive elements and don't make sense in the context of a printed page.

In this PR, we are going to hide the "share this" section in the "print" view. 

I have added some screenshots to explain this. Consider the post below:

<img width="1119" alt="Screenshot 2021-11-30 at 4 03 30 PM" src="https://user-images.githubusercontent.com/6594561/144054770-9e8b7c22-2e6a-4e07-85c2-455e14deb399.png">

If one were to print this view, this is how it would look. 

<img width="627" alt="Screenshot 2021-11-30 at 4 04 41 PM" src="https://user-images.githubusercontent.com/6594561/144054843-90ab204f-2a8f-43e5-8036-41ee9f42dcc7.png">

How we want it to look (this PR makes the changes necessary to make the page look like the image below): 

<img width="653" alt="Screenshot 2021-11-30 at 4 05 56 PM" src="https://user-images.githubusercontent.com/6594561/144054914-04071e04-8cbe-4cf3-b2ca-8dbdd8a84773.png">


#### Changes proposed in this Pull Request:
<!--- Explain what functional changes your PR includes -->
* Added a CSS media query that hides the CSS class `sharedaddy.sd-sharing-enabled` in the print view.

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
 - Go to Jetpack->Settings->Sharing.
 - Make sure "Add sharing buttons to your posts and pages" is enabled
 - Create a post, go to its detailed view and print it (you can press cmd+p to print on a Mac)
 - Observe the results and make sure that the "share this" section is not showing up in the print view
 -  Go to Jetpack->Settings->Sharing, click on "Configure Sharing buttons"
 -  Click on edit label text and change the text to whatever you like. 
 -  Change the button style. 
 -  Go to the post detail view again and confirm that the "share this" section doesn't show up.

danielpost commented on projects/plugins/jetpack/modules/likes/style.css:
```diff
@@ -217,3 +217,10 @@ div.sharedaddy.sd-like-enabled .sd-like .post-likes-widget {
 	display: none;
 }
+
+/* Hide the share this section in the post print view */
+@media print {
+	div.sharedaddy.sd-sharing-enabled {
+	   visibility: hidden;
+	}
+ }
```
Couple thoughts:

1) Is there a specific reason you used `visibility: hidden` instead of `display: none`? `visibility: hidden` leaves an empty space where the sharing buttons used to be, which is probably not what you want.
2) Spacing is a bit off—looks like the actual CSS rule uses spaces instead of tabs.
3) Nitpicky, but you probably don't need the `div`. Just `.sharedaddy.sd-sharing-enabled` should be enough.

spsiddarthan replied:
Thanks for your comments, @danielpost. I have updated the screenshots as well, I can see that the extra space is removed. 

1.  I have changed `visibility: hidden` to `display: none`. I didn't know that `visibility:hidden` leaves an empty space, my CSS isn't the best, will learn. :) 
2. Fixed the spacing.
3. Removed the div. prefix, I assumed that it was an A8C pattern to do it with the prefix, but doesn't appear to be.

danielpost commented on projects/plugins/jetpack/modules/likes/style.css:
```diff
@@ -212,8 +212,14 @@ div.sharedaddy.sd-like-enabled .sd-like .post-likes-widget {
 /* Hide G+ title */
 .sd-gplus .sd-title {
 	display: none;
+	
+}
+
+@media print {
+	.sharedaddy.sd-sharing-enabled{
+		display: none;
+	}
```
```suggestion
}

@media print {
	.sharedaddy.sd-sharing-enabled {
		display: none;
	}
```
I think you added some unintended whitespace. Looks good otherwise!

jeherve reviewed (changes requested):
Your changes update the Likes' CSS, but your screenshots and PR description mention the Sharing module. It may be good to update the description accordingly?

spsiddarthan commented:
Thank you for pointing out that I updated the Likes module's CSS, @jeherve. 

Instead of updating the description, I have moved my changes to the `projects/plugins/jetpack/modules/sharedaddy/sharing.css` file, tested my changes again on the localhost and the sandbox since my changes are related to the sharing functionality.

jeherve reviewed (approved):
This looks good and tests well, it should be good to go! 🚢

kraftbj commented:
r236038-wpcom
